content/blogposts/2024/announcing-prompt-library.md

```diff
@@ -0,0 +1,149 @@
+
+Alternatively, return to Prompt Library, find your prompt, and copy a permalink to share it with others.
+
+![image.png](https://sourcegraph.notion.site/image/https%3A%2F%2Fprod-files-secure.s3.us-west-2.amazonaws.com%2Fe7ce844a-fe2e-4102-b77e-e852aee3841b%2F2a0e4315-7842-4859-9610-63400d467344%2Fimage.png?table=block&id=142a8e11-2658-803f-9899-ff4ee05310bc&spaceId=e7ce844a-fe2e-4102-b77e-e852aee3841b&width=2000&userId=&cache=v2)
+
+# Example Use Cases
+
+Let’s take a look at three realistic example use cases here.
+
+## 1. “Cody is not giving me code in the right language/framework version”
+
+Imagine you are working with a legacy Java application that uses an older version of the Spring framework. You might find that Cody is generating code snippets using a newer version of Spring, which is incompatible with your project. For instance, your project is using Spring 3.x, but Cody provides examples using Spring Boot 2.x features, which are not available in your current setup.
+
+```java
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@RestController
+public class LegacyService {
+
+    @RequestMapping("/legacy")
+    public String handleRequest() {
+        // This method should use older Spring 3.x annotations and configurations
+        return "Handling request with legacy Spring version";
+    }
+}
+```
+
+To address this, you can create a prompt in the Prompt Library that specifies the framework version you are using. For example:
+
+"Please generate Java code that is compatible with the following specifications:
+
+- Java version: 1.6
+- Spring framework version: 3.x
+- Testing framework: JUnit 4"
+
+By sharing this prompt with your team, everyone can ensure that Cody generates code compatible with your legacy system.
+
+## 2. “Cody is not following my organization’s style guide”
+
+Suppose your organization has a specific Python coding style guide that mandates the use of snake_case for variable names and requires docstrings for all functions. However, Cody might generate code that doesn't adhere to these conventions, leading to inconsistencies in your codebase.
+
+```python
+def processData(inputData):
+    """Process the input data and return the result."""
+    processedData = inputData * 2  # This should be snake_case
+    return processedData
+```
+
+To solve this, you can create a comprehensive prompt that instructs Cody to follow your organization's style guide:
+
+"Generate Python code that adheres to our organization's style guide, which includes the following requirements:
+
+- Use snake_case for all variable and function names.
+- Include docstrings for all functions, describing their purpose and parameters.
+- Follow PEP 8 guidelines for code formatting, including indentation and line length."
+
+This prompt can be shared with your colleagues to ensure consistent code style across the team.
```

Comment:
Same as example 1, I think showing the prompt in action would be valuable so users understand what else they need to do to utilize the prompt.